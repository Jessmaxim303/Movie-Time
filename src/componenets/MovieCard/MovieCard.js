import React from 'react';
import { Link } from 'react-router-dom';
import './MovieCard.scss'

export const MovieCard = ({title, posterPath, id, averageRating, userRating}) => {
  console.log(userRating);
  if (!userRating) {
    return (
      <Link className='movie-card' id={id} to={`/movies/${id}`}>
        <img className='poster-img' alt={`movie poster for ${title}`} src={posterPath} />
        <h1 className='movie-card-title'>{title}</h1>
        <footer className='ratings-footer'>
          <h2 className='movie-rating'>{`Avg. Rating: ${averageRating.toFixed(2)}/10`}</h2>
        </footer>
      </Link>
    )
  } else {
    return (
      <Link className='movie-card' id={id} to={`/movies/${id}`}>
        <img className='poster-img' alt={`movie poster for ${title}`} src={posterPath} />
        <h1 className='movie-card-title'>{title}</h1>
        <footer className='ratings-footer'>
          <h2 className='movie-rating'>{`Avg. Rating: ${averageRating.toFixed(2)}/10`}</h2>
          <h2 className='movie-rating user-rating'>{`Your Rating: ${userRating.rating}/10`}</h2>
        </footer>
      </Link>
    )
  }
}
