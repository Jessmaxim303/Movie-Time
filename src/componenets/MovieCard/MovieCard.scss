.movie-card {
  border-radius: 3%;
  border: 3px solid #A8A8A8;
  background-color: #474747;
  width: 15%;
  padding: 2%;
  margin: 2%;
  display: grid;
  font-family: 'Nunito', sans-serif;
}

.poster-img {
  width: 100%;
  height: auto;
  border-radius: 2%;
}

.movie-card-title {
  text-align: center;
  font-size: 2rem;
  color: white;
  margin: 5%;
}

.ratings-footer {
  display: grid;
  width: 100%;

}

.movie-rating {
  justify-self: end;
  padding: 1.5%;
  align-self: end;
  font-weight: 900;
  font-size: 1.2rem;
  border-top: 2px solid #008F11;
  border-bottom: 2px solid #008F11;
  border-radius: 5%;
  color: white;
}
