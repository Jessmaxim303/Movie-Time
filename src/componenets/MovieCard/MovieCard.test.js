import React from 'react';
import { shallow } from 'enzyme';
import { MovieCard } from './MovieCard.js';

describe('MovieCard', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MovieCard />)
  })
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})