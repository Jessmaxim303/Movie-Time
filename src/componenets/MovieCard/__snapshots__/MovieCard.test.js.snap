// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match a snapshot 1`] = `
<Link
  className="movie-card"
  to="/movies/undefined"
>
  <img
    alt="movie poster for undefined"
    className="poster-img"
  />
  <h1
    className="movie-card-title"
  />
  <footer
    className="ratings-footer"
  >
    <h2
      className="movie-rating"
    >
      Avg. Rating: 6.46/10
    </h2>
  </footer>
</Link>
`;
